internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -836,6 +842,35 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	return resourceRead(ctx, d, meta)
 }
 
+func isUpdateAllowed(d *schema.ResourceData) (bool, error) {
```

Comment:
nit: maybe this function can return a boolean, and we handle all messaging outside of the function. Not it is a mix between message in the error + additional text.

Reply:
I added it like that so that `isUpdateAllowed` can return error as per the specific failed criteria

Reply:
Ohh I see, no strong opinion but maybe keeping this function specific to this validation (`isTransitioningFromNewToOldShardSchema`) could make it easier to read. Feel free to ignore.